### PR TITLE
Added Node Stats api changes for the Point in time

### PR DESCRIPTION
**Pull request by ajaymovva**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by bharath-techie** [comment]
Lets fix the build . @ajaymovva

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java [hunk 1]
```diff
@@ -146,6 +161,11 @@ public void add(Stats stats) {
             suggestCount += stats.suggestCount;
             suggestTimeInMillis += stats.suggestTimeInMillis;
             suggestCurrent += stats.suggestCurrent;
+
+            pitCount += stats.pitCount;
+            pitTimeInMillis += stats.pitTimeInMillis;
+            // need consider the count of the pit's current scroll
```
Comment (bharath-techie) [inline server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:167]: nit : fix the comment

File: server/src/test/java/org/opensearch/search/CreatePitMultiNodeTests.java [hunk 2]
```diff
@@ -56,6 +58,14 @@ public void testPit() throws Exception {
             .get();
         assertEquals(2, searchResponse.getSuccessfulShards());
         assertEquals(2, searchResponse.getTotalShards());
+        IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
```
Comment (bharath-techie) [inline server/src/test/java/org/opensearch/search/CreatePitMultiNodeTests.java:61]: same code repeated in all tests, lets move it to a common method and reuse.

File: server/src/test/java/org/opensearch/search/CreatePitSingleNodeTests.java [hunk 3]
```diff
@@ -72,7 +75,23 @@ public void testCreatePITSuccess() throws ExecutionException, InterruptedExcepti
 
         SearchService service = getInstanceFromNode(SearchService.class);
         assertEquals(2, service.getActiveContexts());
+
+        IndicesService indicesService = getInstanceFromNode(IndicesService.class);
+        IndexService indexService = indicesService.indexServiceSafe(resolveIndex("index"));
```
Comment (bharath-techie) [inline server/src/test/java/org/opensearch/search/CreatePitSingleNodeTests.java:80]: move common code to a separate method and reuse

File: server/src/test/java/org/opensearch/search/DeletePitMultiNodeTests.java [hunk 4]
```diff
@@ -70,6 +72,12 @@ public void testDeletePit() throws Exception {
         execute = client().execute(CreatePitAction.INSTANCE, request);
         pitResponse = execute.get();
         pitIds.add(pitResponse.getId());
+        IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
```
Comment (bharath-techie) [inline server/src/test/java/org/opensearch/search/DeletePitMultiNodeTests.java:75]: Same comment as above, reuse duplicate code

File: server/src/test/java/org/opensearch/search/SearchServiceTests.java [hunk 5]
```diff
@@ -1431,13 +1431,22 @@ public void testDeletePitReaderContext() {
         contextIds.add(pitSearchContextIdForNode);
 
         assertThat(searchService.getActiveContexts(), equalTo(1));
+        IndicesService indicesService = getInstanceFromNode(IndicesService.class);
```
Comment (bharath-techie) [inline server/src/test/java/org/opensearch/search/SearchServiceTests.java:1434]: Same comment as above, reuse duplicate code

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java (hunk 1 above)
Comment (bharath-techie) [inline server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:167]: nit : fix  comment

**Comment by dreamer-89** [comment]
@ajaymovva @bharath-techie : Are you still working on this ?

**Comment by dreamer-89** [comment]
@ajaymovva @bharath-techie: Closing this stale PR due to no activity. Please feel free to reopen or create a new one if needed.

**Pull request closed without merging**